utils/git_export_campaign

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+
+"""A command similar to git-subtree, for exporting the history of a campaign.
+
+Hardcoded for the export of An Orcish Incursion, but added to the utils/ dir
+as other campaigns will likely be removed before 1.16. The result is a Git
+branch containing just that campaign including translations, combined with a
+copy of the GPL.
+"""
+
+# Hardcode which campaign to export, no need to handle command-line options in
+# such a rarely-run script.
+CAMPAIGN_NAME = 'An_Orcish_Incursion'
+CAMPAIGN_TEXTDOMAIN = 'wesnoth-aoi'
+CAMPAIGN_PATHS = ['data/campaigns/' + CAMPAIGN_NAME, 'po/' + CAMPAIGN_TEXTDOMAIN]
+
+import os, subprocess, sys
+
+def format_mktree_subdir(sha, filename):
+    """Returns the line for `git mktree -z` to add a subdirectory named `filename`, with contents `sha`."""
+    line = b'040000 tree ' + sha + b'\t' + bytes(filename, encoding='utf-8') + b'\0'
+    return line
+
+def format_mktree_file(sha, filename):
+    """Returns the line for `git mktree -z` to add a file named `filename`, with contents `sha`."""
+    line = b'100644 blob ' + sha + b'\t' + bytes(filename, encoding='utf-8') + b'\0'
+    return line
+
+# A mapping which maps from the existing branch's SHA1 ids to the corresponding
+# commit on the rewritten branch.
+old_to_new_objects = {}
+
+def create_root_tree(from_commitish):
+    """The contents of the filesystem for each commit, returns the SHA1 tree-ish."""
+    try:
+        data_campaign_tree = subprocess.run(
+            ['git', 'ls-tree', '-z', from_commitish + ':data/campaigns/' + CAMPAIGN_NAME + '/'],
+            check=True, capture_output=True).stdout
+    except subprocess.CalledProcessError:
+        # This must have been a commit that deleted all of the campaign files
+        data_campaign_tree = b''
+
+    # In the newly created tree, add translations/CAMPAIGN_TEXTDOMAIN/*.po and the .pot
```

Comment:
Need to handle the situation that both data/campaigns/.../translations and po/wesnoth-.../ exist, as happened in d524ef29c72917c8ac74d4f898bb7f842520a85c. The low-level git commands being used allow that, but it means `git fsck` will complain afterwards.